src/rmm.cpp

```diff
@@ -59,39 +59,48 @@ const char * rmmGetErrorString(rmmError_t errcode) {
 // Initialize memory manager state and storage.
 rmmError_t rmmInitialize(rmmOptions_t *options)
 {
+  rmm::Manager::getInstance().initialize(options);
+
+  rmm::mr::device_memory_resource * memory_resource = nullptr;
+
+  if (rmm::Manager::usePoolAllocator()) {
+    if (rmm::Manager::useManagedMemory()) {
+      memory_resource = new rmm::mr::cnmem_managed_memory_resource{
+        rmm::Manager::getOptions().initial_pool_size};
+    } else {
+      memory_resource = new rmm::mr::cnmem_memory_resource{
+        rmm::Manager::getOptions().initial_pool_size};
+    } 
+  } else if (rmm::Manager::useManagedMemory()) {
+    memory_resource = new rmm::mr::managed_memory_resource();
+  }
 
-    rmm::Manager::getInstance().initialize(options);
-    rmm::mr::device_memory_resource * memory_resource = rmm::mr::detail::initial_resource();
-
-    if (rmm::Manager::usePoolAllocator())
-    {
-        memory_resource = rmm::mr::detail::pool_resource(rmm::Manager::getOptions().initial_pool_size);
-
-    }else if(rmm::Manager::useManagedMemory()){
-      memory_resource = rmm::mr::detail::managed_resource();
-
-    }else if(rmm::Manager::useManagedMemory() && rmm::Manager::usePoolAllocator()){
```

Comment:
Note that this previous logic was incorrect -- there would be no way to create a managed pool allocator.

Reply:
Yeah, I noticed that as well and fixed it in the same way here: https://github.com/rapidsai/rmm/pull/106 :)